File: CloudMusic/src/main/java/com/ken/android/CloudMusic/Adapter/RecyclerViewAdapter.java
package com.ken.android.CloudMusic.Adapter;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.RelativeLayout;
import android.widget.TextView;

import com.ken.android.CloudMusic.FilesRead.ListsInfo;
import com.ken.android.CloudMusic.R;

import org.xutils.view.annotation.ContentView;

import java.util.List;

/**
 * Created by axnshy on 16/8/9.
 */
public class RecyclerViewAdapter extends RecyclerView.Adapter<MyViewHolder> {

    private Context context;
    private List<ListsInfo> mListsList;
    private LayoutInflater mInflater;
    private int screenWidth;
    private int screenHeight;

    public interface OnItemClickListener {
        void onItemClick(View view, int position);

        void onItemLongClick(View view, int position);
    }

    private OnItemClickListener mOnItemClickListener;

    public void setOnItemClickListener(OnItemClickListener listener) {
        this.mOnItemClickListener = listener;
    }

    public RecyclerViewAdapter(Context context, List<ListsInfo> list, int width, int length) {
        super();
        this.context = context;
        this.mListsList = list;
        this.screenHeight = length;
        this.screenWidth = width;
        mInflater = LayoutInflater.from(context);
    }

    @Override
    public MyViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = mInflater.inflate(R.layout.music_list_view, parent, false);
        MyViewHolder viewHolder = new MyViewHolder(view);
        return viewHolder;
    }

    @Override
    public void onBindViewHolder(final MyViewHolder holder, final int position) {
        RecyclerView.LayoutParams params = (RecyclerView.LayoutParams) holder.listAvatarLayout.getLayoutParams();
        Log.w("TAG", "screenWidth--------" + screenWidth + "     screenHeight----------" + screenHeight);
        params.width = (screenWidth - 4 * 8) / 3;
        params.height = (int) (params.width * 1.414);
        holder.listAvatarLayout.setLayoutParams(params);
//        holder.listCountTx = (TextView) holder.findViewById(R.id.tv_musiclist_name);
//        holder.listNameTx = (TextView) itemView.findViewById(R.id.tv_musiclist_count);
        holder.listNameTx.setText(mListsList.get(position).getListName());
        holder.listCountTx.setText(mListsList.get(position).getListCount() + "");
        if (mOnItemClickListener != null) {
            holder.itemView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View view) {
                    mOnItemClickListener.onItemClick(holder.itemView, position);
                }
            });
            holder.itemView.setOnLongClickListener(new View.OnLongClickListener() {
                @Override
                public boolean onLongClick(View view) {
                    mOnItemClickListener.onItemLongClick(holder.itemView,position);
                    return false;
                }
            });
        }

    }

    @Override
    public int getItemCount() {
        return mListsList.size();
    }
}

//@ContentView(R.layout.music_list_view)
class MyViewHolder extends RecyclerView.ViewHolder {

    RelativeLayout listAvatarLayout;
    TextView listNameTx;
    TextView listCountTx;

    public MyViewHolder(View itemView) {
        super(itemView);
        listAvatarLayout = (RelativeLayout) itemView.findViewById(R.id.layout_list_avatar);
        listNameTx = (TextView) itemView.findViewById(R.id.tv_musiclist_name);
        listCountTx = (TextView) itemView.findViewById(R.id.tv_musiclist_count);
    }
}

File: SmartPlay/src/main/java/com/luffy/smartplay/ui/adapter/RecyclerViewAdapter.kt
package com.luffy.smartplay.ui.adapter

import android.graphics.drawable.Drawable
import android.view.LayoutInflater
import android.view.ViewGroup
import com.luffy.smartplay.R
import com.luffy.smartplay.databinding.MusicListViewBinding
import com.luffy.smartplay.db.bean.AlbumData
import com.luffy.smartplay.ui.base.BaseBindingAdapter
import com.luffy.smartplay.ui.base.BindingViewHolder

/**
 * Created by axnshy on 16/8/9.
 */
class RecyclerViewAdapter(val list: List<AlbumData> = mutableListOf()) : BaseBindingAdapter<MusicListViewBinding>() { //@ContentView(R.layout.music_list_view)

    override fun onBindViewHolder(holder: BindingViewHolder<MusicListViewBinding>, position: Int) {
        super.onBindViewHolder(holder, position)
        holder.holder.apply {
            tvMusiclistName.text = list[position].albumName
            tvMusiclistCount.text = list[position].count.toString()

            layoutListAvatar.background = if (list[position].albumAvatar != null) {
                Drawable.createFromPath(list[position].albumAvatar)
            } else holder.itemView.context.resources.getDrawable(R.drawable.h1)
        }

    }

    override fun createViewHolder(
        layoutInflater: LayoutInflater,
        parent: ViewGroup,
        viewType: Int
    ): MusicListViewBinding {
        return MusicListViewBinding.inflate(layoutInflater,parent,false)
    }

    override fun getItemCount(): Int {
        return list.size
    }

}